### DISCUSSION TITLE
Congo for Statiq....

### DISCUSSION THREAD

**Discussion by harperjohn** [statiqdev/Discussions#201] [Q&A]
BLUF:

I would love to have/create a Congo-like theme for Statiq. Specifically the integration of Tailwind and out of the box "Dark Mode", and a clean and "general enough" theme to use as is, and/or to easily extend/customize. 

https://jpanther.github.io/congo/

Details:

Hugo is really nice... and Go seems like a great language/ecosystem but like most people here, I love and am committed to the .Net world!

I am not a Hugo expert, and was (and still am to some extent) overwhelmed by all the details of Hugo, and I feel some of that same "overwhelm" as I start to learn Statiq.

But, I have to say that using and learning the Congo theme was an excellent doorway into Hugo for me. In general, using a Theme as a starting point is nice because, by its nature it has a practical/usable aspect, and it focuses on the set of features to implement that practicality. 

And, as in the case of Congo, it has a great sample website that shows how to use the Theme, with the excellent side effect of demonstrating/teaching many aspects of Hugo.

So I would like to the something similar with Statiq.... that is... create a Congo like theme *and* fully document it in a sample website. 

Has anyone done something similar to this already?

Any helpful thoughts/starting-points/guidance?

Does this seem reasonable/desirable?

**Comment by daveaglick** [comment]
I think that's a fantastic idea. I spent a lot of time reworking stuff in Statiq Web to make it easier to create and distribute themes over how Wyam did it (Wyam is a predecessor of sorts to Statiq, though mostly spiritually at this point since nearly all the code has been rewritten over the years). Of course after doing all the groundwork I got distracted and never actually _wrote_ any additional themes 😂.

There is a good example in [CleanBlog](https://github.com/statiqdev/CleanBlog) though. From the way to package a theme, to using a theme-specific `settings.yml` file, to using the [archive feature](https://www.statiq.dev/guide/web/archives) for the [home page](https://github.com/statiqdev/CleanBlog/blob/main/input/index.cshtml), it's got most of the stuff you'd want to replicate when writing a new theme. I've worked on a few custom themes in Statiq and I almost always start with CleanBlog, rip out most of the stuff in [_layout.cshtml](https://github.com/statiqdev/CleanBlog/blob/main/input/_layout.cshtml) and the partials, and then start adding back in stuff from the theme I'm building. That pattern seems to work pretty well.

**Reply by reyespinosa1996** [reply]
After 3 days of cracking my head just to try to found where to start with CleanBlog in order to customize it to my own needs, I just found this. There should be something like this in the docs...

**Comment by harperjohn** [comment]
As we know... the best way to "eat an elephant"... is ... well... "one bite at a time" :)

I have done the getting started exercise. This is very simple, but it was nice to show the "build process" and nice to see the auto refresh "dev server", etc.  I added a little javascript and a button to see how it handled the embedded html and script, and that seemed to work fine (I love Blazor, as it gives us a partial relief from the dreaded javascript... but with javascript, while I am still "not a fan", I have faced and accepted it as a (current) necessary evil. 

So I guess the next thing will be to give CleanBlog a try.

Yes - it was disappointing to find out that there was not a huge gallery of themes.... but from what I have done with Hugo so far, I have found that I have gravitated towards a very small set of Themes (mostly just Congo now) and use them/it as generic foundation. 

One of the first things that I would like to do is to integrate Tailwindcss with Statiq. (from now on when I say "Statiq", I will most likely mean Statiq and my derivation of CleanBlog)

I have found this: https://www.ankursheel.com/blog/add-tailwind-css-statiq-website

But it is coming up on two years old, so I was wondering if there were any newer notes on the topic.

Oh - and by the way - I expect that (at least at first) some/most of my questions will have at least partial answers that I just need you to point me to. And then as time goes on, I might start breaking some newer ground and will then need a bit more explanation - but until that happens I will be more than glad to read whatever I am pointed too.

**Reply by daveaglick** [reply]
> One of the first things that I would like to do is to integrate Tailwindcss with Statiq. (from now on when I say "Statiq", I will most likely mean Statiq and my derivation of CleanBlog)

Take a look at the [docs site for Spectre Console](https://github.com/spectreconsole/spectre.console/tree/main/docs) - they've done an amazing job writing a bespoke Statiq theme with all kinds of cool Tailwind and other integration.

**Reply by harperjohn** [reply]
Wow @daveaglick !!! 

I just took a quick look, but at first glance it seems to have covered a bunch of what I had in mind!!!! My first/current need is for a project documentation site, so this seems to be a PERFECT example!!!! With TailwindCss and darkmode as a bonus!!!!

This seems to be a GREAT head start! I will dive into this as my time permits and report back here as I make progress!!!

Thanks Dave!!!!! :)
